pkg/sql/opt/norm/rules/limit.opt

```diff
@@ -14,6 +14,19 @@
 =>
 $input
 
+# EliminateLimitInExists discards a Limit operator with a positive limit inside
+# an Exist operator.
+[EliminateLimitInExists, Normalize]
+(Exists
```

Comment:
Should this be more general? E.g. should it match SemiJoin and AntiJoin patterns instead of only matching exists specifically?

Reply:
This also doesn't restrict the normalization to only applying inside a `WHERE` clause, which maybe isn't desired? For example, should the limit not be eliminated in something like:

```
SELECT EXISTS(SELECT * FROM (VALUES ('radu', 'andy')) LIMIT 1)
```